### ISSUE TITLE
xeyes on Windows 10 WSL 2 and Debian: "Can't open display:"

### ISSUE THREAD

**Issue by burnbasket** [MicrosoftDocs/WSL#1669]
### Documentation Issue

https://learn.microsoft.com/en-us/windows/wsl/tutorials/gui-apps

The documentation states in Prerequisites that I need "Windows 10 Build 19044+" and the vGPU driver. It later states in a note "only supported with WSL 2".

This is my environment:
```
ver
Microsoft Windows [Version 10.0.19044.2251]

wsl --list --all --verbose
  NAME      STATE           VERSION
* Debian    Running         2
```
I did an "wsl --update" and a reboot before executing the following on the Debian terminal:
```
> echo $DISPLAY

> xeyes
Error: Can't open display:
> export DISPLAY=:0.0
> xeyes
Error: Can't open display: :0.0
```

I tried following the advice on [Diagnosing "cannot open display" type issues with WSLg](https://github.com/microsoft/wslg/wiki/Diagnosing-%22cannot-open-display%22-type-issues-with-WSLg) although I'm not sure this applies, because it purports to require "Windows build 21364+".

I have none of these:
/mnt/wslg/.X11-unix
/tmp/.X11-unix

I extrapolated in my installation the first of them would be in
/mnt/wsl/.X11-unix
This doesn't exist. Creating an empty symlink worked but didn't help:
```
> ln -s /mnt/wsl/.X11-unix /tmp/
> ll /tmp/.X11-unix
lrwxrwxrwx 1 username groupname 18 Nov 30 09:40 /tmp/.X11-unix -> /mnt/wsl/.X11-unix
> xeyes
Error: Can't open display: :0.0
```
Now I'm out of ideas.

### Link to documentation page

https://learn.microsoft.com/en-us/windows/wsl/tutorials/gui-apps

### Suggested Improvements

Maybe more detail is required?

**Comment by guitarpicker** [comment]
Additional info: there is conflicting documentation on the [troubleshooting page](https://github.com/microsoft/wslg/wiki/Diagnosing-%22cannot-open-display%22-type-issues-with-WSLg) linked by the aforementioned page.  It states "Verify you are running on Windows build 21364+" which is probably the reason that this doesn't work on 19044.

**Comment by guitarpicker** [comment]
I'm pretty sure that this is a problem of incorrect documentation.  I suspect that Windows build 21364+ is the actual requirement, not 19044+.  I don't have the Insider build to test on, though.  This works great on Windows 11, and I was hoping to get it working on 10.

**Comment by burnbasket** [comment]
Thanks @m3lab, for narrowing down the problem.

I don't have KB5020030 installed on the machine concerned. Unless somebody else does so first, I will report the results once I get back on that computer, which might be a couple of days.

**Comment by burnbasket** [comment]
I did update to @m3lab's suggested KB5020030. This got me part of the way there.

My WSL installation still didn't use the most recent kernel.

This can be verified by running:
`wsl --version`
In my case that still showed "Kernel Version: 5.10.something".

This can be remedied by installing the latest release of  "Microsoft.WSL_*.msixbundle".
It can be found and downloaded here:
[https://github.com/microsoft/WSL/releases](https://github.com/microsoft/WSL/releases)
After that the WSL kernel was updated to the most recent one by running
`Add-AppPackage .\Downloads\Microsoft.WSL_1.0.3.0_x64_ARM64.msixbundle`
where the string "1.0.3.0_x64_ARM64" needs to be replaced with the current version.

Having done that verifying the kernel version by running
`wsl --version`
showed "Kernel Version: 5.17.something".

WSL GUI programs on Windows 10 now work as expected.
